### ISSUE TITLE
Wrap expensive log statements in isLoggable check

### ISSUE THREAD

**Issue by burkedavison** [googleapis/sdk-platform-java#2196]
String manipulation that occurs before invoking the log method:
* https://github.com/googleapis/sdk-platform-java/blob/7fc555c20c642e813ee60fc27e05966143afc924/gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java#L290-L301

And a method that performs relatively complex logic (env var checking, file reading) before choosing what to log:
* https://github.com/googleapis/sdk-platform-java/blob/7fc555c20c642e813ee60fc27e05966143afc924/gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java#L275
* The first check in this method could check the log level and exit early if no logging will be performed.

A quick search of other log uses in sdk-platform-java and google-cloud-java did not show any other substantial improvements.

**Comment by suztomo** [comment]
Is that part called many times in a second? If not, I wouldn't care about these unnecessary string concatenation.

**Comment by blakeli0** [comment]
> Is that part called many times in a second? If not, I wouldn't care about these unnecessary string concatenation.

I agree. That part should only happen during channel resizing, which happens infrequently.

@burkedavison What do you mean by `String manipulation that occurs before invoking the log method`? Based on the referenced code, I think it only occurs if certain conditions met?
For the second scenario, it only happens during client initialization so I wouldn't worry about it either. In addition, the `logDirectPathMisconfig()` will be refactored relatively soon so that all the checks should be only performed once during client initialization.

**Comment by burkedavison** [comment]
This came on my radar after a conversation about native images' optimization bottlenecks. (ex: Seem really unfortunate to open a file during cold start just to ignore the result.)

But more generally, the cost to apply a consistent rule seems a lot lower than repeatedly reasoning about each instance within the overall application context. 

Here's our Sonar configuration complaining about it: https://sonarcloud.io/project/issues?cleanCodeAttributeCategories=INTENTIONAL&fileUuids=AYUdBPg5FPHUkXE9R93u&resolved=false&id=googleapis_gapic-generator-java&open=AYUdBZlBFPHUkXE9R_NM

Probably not worth a long discussion if you'd rather close it.
